modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/VirtualFlowTest.java

```diff
@@ -1207,17 +1226,178 @@ public void testSheetChildrenRemainsConstant() {
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testGradualMoveForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        vf.resize(600,400);
+        ScrollBar sb = vf.shim_getVbar();
+        double s0 = sb.getLayoutY();
+        double s1 = s0;
+        double position = vf.getPosition();
+        double newPosition =0d;
+        double delta = 0;
+        double newDelta = 0;
+        vf.layout();
+        for (int i = 0; i < 50; i++) {
+            vf.scrollPixels(10);
+            vf.layout();
+            newPosition = vf.getPosition();
+            s1 = sb.getLayoutY();
+            newDelta = newPosition - position;
+            System.err.println("s0 = "+s0+", s1 = "+s1);
+            System.err.println("newDelta = "+newDelta+", delta = "+delta);
+            if (i > 0) {
+                double diff = Math.abs((newDelta-delta)/newDelta);
+                System.err.println("diff = "+diff);
+                // maximum 10% difference allowed
+                assertTrue("Too much variation while scrolling (from "+s0+" to "+s1+")", diff < 0.1);
+            }
+            System.err.println("S1 = "+s1);
+            System.err.println("pos = "+vf.getPosition());
+            assertFalse("Thumb moving in the wrong direction at index ", s1 < s0);
+            s0 = s1;
+            delta = newDelta;
+            position = newPosition;
+        }
+    }
+}
+
+class GraphicalCellStub extends IndexedCellShim<Node> {
+    static ArrayList<Circle> circleList = new ArrayList<>();
+    static {
+        circleList.add(new Circle(10));
+        circleList.add(new Circle(20));
+        circleList.add(new Circle(100));
+        circleList.add(new Circle(30));
+        circleList.add(new Circle(50));
+        circleList.add(new Circle(200));
+        circleList.add(new Circle(60));
+    }
+
+    private int idx = -1;
+    Node myItem = null;
+
+    public GraphicalCellStub() { init(); }
+
+    private void init( ) {
+        System.err.println("Init vf cell "+this);
+        setSkin(new SkinStub<GraphicalCellStub>(this));
+    }
+
+    @Override
+    public void updateItem(Node item, boolean empty) {
+        super.updateItem(item, empty);
+        if (empty || item == null) {
+            setText(null);
+            setGraphic(null);
+        } else {
+            setGraphic(item);
+        }
+    }
+
+    @Override
+    public void updateIndex(int i) {
+        super.updateIndex(i);
+        if ((i > -1) && (circleList.size() > i)) {
+            this.idx = i;
+            updateItem(circleList.get(i), false);
+        } else {
+            updateItem(null, true);
+        }
+    }
+
+    @Override
+    protected double computePrefHeight(double width) {
+        double answer = super.computePrefHeight(width);
+        if ((idx > -1) && (idx < circleList.size())){
+            answer = 2* circleList.get(idx).getRadius()+ 6;
```

Comment:
Minor: spacing is a little inconsistent for these two lines.

Reply:
changed